src/security/body_parse/header.cpp

```diff
@@ -0,0 +1,520 @@
+      return false;
+    }
+  }
+  return true;
+}
+
+inline std::string to_lc(std::string_view sv) {
+  std::string result;
+  result.reserve(sv.size());
+  std::transform(sv.begin(), sv.end(), std::back_inserter(result),
+                 [](unsigned char c) { return std::tolower(c); });
+  return result;
+}
+
+void consume_ows(std::string_view &sv) {
+  while (!sv.empty() && (sv.front() == ' ' || sv.front() == '\t')) {
+    sv.remove_prefix(1);
+  }
+}
+
+/*
+ * Consumes a token according to RFC 9110, section 5.6.2.
+ *
+ * https://httpwg.org/specs/rfc9110.html#rfc.section.5.6.2
+ *   token          = 1*tchar
+ *   tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
+ *                  / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+ *                  / DIGIT / ALPHA
+ *                  ; any VCHAR, except delimiters
+ *
+ * For multipart/form-data, RFC 9110 refers to RFC 7578 ("Returning Values from
+ * Forms: multipart/form-data"), which in refers to RFC 2183 ("The
+ * Content-Disposition Header Field"), which in turn defines their tokens like
+ * this:
+ *
+ * https://datatracker.ietf.org/doc/html/rfc2045 (by reference to RFC 822)
+ *   token      := 1*<any (US-ASCII) CHAR except SPACE, CTLs,
+ *                    or tspecials>
+ *
+ *   tspecials :=  "(" / ")" / "<" / ">" / "@" /
+ *                 "," / ";" / ":" / "\" / <">
+ *                 "/" / "[" / "]" / "?" / "="
+ *                 ; Must be in quoted-string,
+ *                 ; to use within parameter values
+ *
+ * This is both more permissive (allows {}) and more restrictive (forbids
+ * characters outside ASCII).
+ */
+std::optional<std::string_view> consume_wg_token(std::string_view &sv) {
+  static constexpr std::string_view tchar =
+      "abcdefghijklmnopqrstuvwxyz"
+      "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+      "0123456789!#$%&'*+-.^_`|~";
+  auto end = sv.find_first_not_of(tchar);
+  if (end == 0) {
+    return std::nullopt;
+  }
+  if (end == std::string_view::npos) {
+    end = sv.size();
+  }
+  auto ret = std::optional{sv.substr(0, end)};
+  sv.remove_prefix(end);
+  return ret;
+}
+
+/*
+ * https://httpwg.org/specs/rfc9110.html#quoted.strings
+ * quoted-string  = DQUOTE *( qdtext / quoted-pair ) DQUOTE
+ * qdtext         = HTAB / SP / %x21 / %x23-5B / %x5D-7E / obs-text
+ * obs-text       = %x80-FF
+ * quoted-pair    = "\" ( HTAB / SP / VCHAR / obs-text )
+ */
+std::optional<std::string> consume_9110_quoted_string(std::string_view &sv) {
+  if (sv.front() != '"') {
+    return std::nullopt;
+  }
+  sv.remove_prefix(1);
+
+  std::string result;
+  while (!sv.empty()) {
+    unsigned char ch = sv.front();
+    sv.remove_prefix(1);
+    if (ch == '"' /* 0x22 */) {
+      return result;
+    }
+
+    // qdtext
+    if (ch == '\t' || ch == ' ' || (ch >= 0x21 && ch != '\\' && ch != 0x7F)) {
+      result.push_back(ch);
+      continue;
+    }
+
+    // quoted-pair
+    if (ch == '\\') {
+      if (sv.empty()) {
+        return std::nullopt;
```

Comment:
More readable to just break, here and at the end of the loop.

Reply:
there may be other advantages to avoid early returns, but I don't think readability is one them. This way you know right there what is happening, rather than scrolling down to follow the flow and find out

Reply:
It's effectively the same argument as goto cleanup.